docs/platforms/javascript/common/sourcemaps/uploading/typescript.mdx

```diff
@@ -7,4 +7,60 @@ notSupported:
   - javascript.sveltekit
 ---
 
-<PlatformContent includePath="sourcemaps/upload/typescript" />
+In this guide, you'll learn how to successfully upload source maps for TypeScript using our `sentry-cli` tool.
+
+<Include name="debug-id-requirement.mdx" />
+
+<Note>
+
+This guide is only applicable if you're using `tsc` to compile your project. If you use another tool (such as webpack) in combination with TypeScript, you'll most likely want to follow that guide instead.
+
+</Note>
+
+## Automatic Setup
+
+The easiest way to configure uploading source maps with `tsc` and `sentry-cli` is by using the Sentry Wizard:
+
+<Include name="sourcemaps-wizard-instructions.mdx" />
+
+If you want to configure source maps upload manually instead, follow the steps below.
+
+## Manual Setup
+
+### 1. Generate Source Maps
+
+First, configure the TypeScript compiler to output source maps:
+
+```json {filename:tsconfig.json}
+{
+  "compilerOptions": {
+    "sourceMap": true,
+    "inlineSources": true,
+
+    // Set `sourceRoot` to  "/" to strip the build path prefix from
+    // generated source code references. This will improve issue grouping in Sentry.
+    "sourceRoot": "/"
+  }
+}
+```
+
+<Alert>
+  Generating sourcemaps may expose them to the public, potentially causing your
+  source code to be leaked. You can prevent your source maps from being publicly
+  accessible by configuring your server to deny access to `.js.map` files, or by
+  deleting the sourcemaps before deploying your application.
+</Alert>
+
+<Include name="sentry-cli-sourcemaps.mdx" />
+
+## Dealing with TSLib
+
+During compilation, if needed, TypeScript will inject some of its runtime dependencies into the output files it produces. It can include things like polyfills for function generators or some APIs not available in all the environments.
+
+However, this makes it impossible to map frames from compiled code to the original sources, as there are no original sources.
+
+We can still make it work, though. To do this, we need to tell the TypeScript compiler not to inject those code snippets and use its own 3rd party package called `tslib`, which is internally the part of a compiler.
+
+The only things that have to change are inside the TypeScript config file, not your source code.
+
+First, make sure that `tslib` is listed as the dependency in your `package.json` file. Once that's done, add two entries in `compilerOptions` section of your `tsconfig.json`. `"noEmitHelpers": true` and `"importHelpers": true`. That's it. Now, we can correctly map the source maps for all your stack trace frames, including internal TypeScript compiler code snippets.
```

Comment:
```suggestion

During compilation, TypeScript will inject some of its runtime dependencies into the output files it produces if needed. It can include things like polyfills for function generators or APIs that may not be available in all the environments. However, the fact that there aren't any original sources makes it impossible to map frames from compiled code to the original sources.

As a workaround, you'll need to tell the TypeScript compiler to use `tslib`, its own 3rd party package, (which is internally the part of a compiler) instead of injecting runtime dependencies. You'll only need to change what's inside the TypeScript config file, not your source code. Here's how:

1. Make sure that `tslib` is listed as the dependency in your `package.json` file. 
2. Once that's done, add two entries in `compilerOptions` section of your `tsconfig.json`:
- `"noEmitHelpers": true` and 
-`"importHelpers": true`. 

That's it! Now you should be able to correctly map the source maps for all your stack trace frames, including internal TypeScript compiler code snippets.
```
Please look over the changes I made here carefully. I want to make sure I didn't accidentally change the meaning of anything.